wazo_amid/config.py

```diff
@@ -172,5 +176,59 @@ def _load_key_file(config: dict[str, Any]) -> dict[str, Any]:
 def load_config() -> AmidConfigDict:
     cli_config = _get_cli_config()
     file_config = read_config_file_hierarchy(ChainMap(cli_config, _DEFAULT_CONFIG))
+    reinterpreted_config = _get_reinterpreted_raw_values(
+        ChainMap(cli_config, file_config, _DEFAULT_CONFIG)
+    )
     service_key = _load_key_file(ChainMap(cli_config, file_config, _DEFAULT_CONFIG))
-    return ChainMap(cli_config, service_key, file_config, _DEFAULT_CONFIG)
+    return ChainMap(
+        reinterpreted_config, cli_config, service_key, file_config, _DEFAULT_CONFIG
+    )
+
+
+def _parse_cli_args(argv: list[str]) -> dict[str, Any]:
```

Comment:
Should this be part of this PR? I feel like it could be its own ticket.
And also, calling the cli while the service is already running raise an error:
```
# wazo-amid 
2024-06-27 03:36:49,752 [1621589] (INFO) (wazo_amid.ami.client): Connecting AMI client to localhost:5038
2024-06-27 03:36:49,754 [1621589] (INFO) (wazo_amid.ami.client): AMI client connected to localhost:5038
2024-06-27 03:36:50,079 [1621589] (INFO) (wazo_amid.ami.client): Disconnecting AMI client. Reason: explicit stop
2024-06-27 03:36:50,080 [1621589] (CRITICAL) (root): No socket could be created -- (('127.0.0.1', 9491): [Errno 98] Address already in use)
Traceback (most recent call last):
  File "/usr/local/bin/wazo-amid", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/usr/src/wazo/wazo-amid/bin/wazo-amid", line 6, in <module>
    main()
  File "/usr/src/wazo/wazo-amid/wazo_amid/bin/daemon.py", line 34, in main
    controller.run()
  File "/usr/src/wazo/wazo-amid/wazo_amid/controller.py", line 48, in run
    self._run_rest_api()
  File "/usr/src/wazo/wazo-amid/wazo_amid/controller.py", line 70, in _run_rest_api
    rest_api.run(self._config['rest_api'])
  File "/usr/src/wazo/wazo-amid/wazo_amid/rest_api.py", line 116, in run
    wsgi_server.start()
  File "/usr/lib/python3/dist-packages/cheroot/server.py", line 1836, in start
    self.prepare()
  File "/usr/lib/python3/dist-packages/cheroot/server.py", line 1791, in prepare
    raise socket.error(msg)
OSError: No socket could be created -- (('127.0.0.1', 9491): [Errno 98] Address already in use)
```

Reply:
That's normal, this cli is to start the daemon process. So running this command will try to bind to the configured port & address, which happens to already be used by the already running wazo-amid.

Reply:
@rbienvenu0 You are completely right. Actually, that function is useless in wazo-amid, since it is never used, and the function `_get_cli_config()` already takes care of parsing the CLI args.